simulator/ns-3.35/src/point-to-point/model/switch-mmu.cc

```diff
@@ -35,39 +35,56 @@ TypeId SwitchMmu::GetTypeId(void) {
 	return tid;
 }
 
+/*
+We model the switch shared memory (purely based on our understanding and experience).
+The switch has an on-chip buffer which has `bufferPool` size. This buffer is shared across all port and queues in the switch.
+
+`bufferPool` is further split into multiple pools at the ingress and egress.
+
+It would be easier to understand from here on if you consider Ingress/Egress are merely just counters. These are not separate buffer locations or chips...!
+
+First, `ingressPool` (size) accounts for ingress buffering shared by both lossy and lossless traffic.
+Second, at the egress, `egressPool[LOSSY]` (size) accounts for buffering lossy traffic at the egress and similarly `egressPool[LOSSLESS]` for lossless traffic.
+*/
+
+
 SwitchMmu::SwitchMmu(void) {
 
+	// Here we just initalize some default values. The buffer can be configured using Set functions through the simulation file later.
+
 	// Buffer pools
 	bufferPool = 24 * 1024 * 1024; // ASIC buffer size i.e, total shared buffer
-	ingressPool = 18 * 1024 * 1024; // ingress lossless pool
-	egressPool[LOSSLESS] = 24 * 1024 * 1024; // egress lossless pool. Lossless bypasses egress admission
-	egressPool[LOSSY] = 14 * 1024 * 1024; // egress lossy pool.
-	egressPoolAll = 24 * 1024 * 1024;
-	xoffTotal = 0; //6 * 1024 * 1024;
-
+	ingressPool = 18 * 1024 * 1024; // Size of ingress pool. Note: This is shared by both lossless and lossy traffic.
+	egressPool[LOSSLESS] = 24 * 1024 * 1024; // Size of egress lossless pool. Lossless bypasses egress admission
+	egressPool[LOSSY] = 14 * 1024 * 1024; // Size of egress lossy pool.
+	egressPoolAll = 24 * 1024 * 1024; // Not for now. For later use.
+	xoffTotal = 0; //6 * 1024 * 1024; // Total headroom space in the shared buffer pool // This value is incremented when SetHeadroom function is used. So setting it to zero initially. Note: This would mean that headroom must be set explicity.
+
+	// aggregate run time
+	ingressTotalUsed = 0;  // IMPORTANT TO NOTE: THIS IS NOT bytes in the "ingress pool". This is the Total bytes in the ingress, which includes occupied buffer in reserved + headroom + ingresspool. 
```

Comment:
I don't know why we define ingressTotalUsed here. We should track ingress shared buffer pool occupancy and PFC headroom buffer occupancy, respectively.

Reply:
My understanding is that, `ingressTotalUsed` accounts for all the bytes buffered at the ingress, including bytes from ingressPool and xoffTotal (headroom). 

We also have xoffTotalUsed which is the total bytes used form headroom.

Using the above two, we can also access the total bytes in the ingressPool specifically: `ingressTotalUsed - xoffTotalUsed`